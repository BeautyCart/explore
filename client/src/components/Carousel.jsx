import React from 'react';
import styled from 'styled-components';

const axios = require('axios');

const Container = styled.div`
  width: 980px;
  /* background: #ccc; */
  min-height: 164px;
  overflow: hidden;
`;

const Track = styled(Container)`
  width: 940px;
  display: flex;
`
const CardBox = styled(Container)`
  width: 188px;
  flex-shrink: 0;
  height: 164px:
  border: 1px solid #aaa;
  padding: 0px 12px;
  box-sizing: border-box;
`
const Card = styled(CardBox)`
  width: 100%;
  height: 100%;
  /* background: blue; */
  border: 1px solid #aaa;
`
const Nav = styled.button`
  width: 40px;
  height: 40px;
  border: 1px solid #ccc;
  border-radius: 50%;
`

class Carousel extends React.Component{
  constructor(props) {
    super(props);
    this.state = {
      users: "loading"
    }
  }

  componentDidMount() {}

  render() {return (
    <Container>
      <Track>
        {console.log(`this.props.users[0] is ${this.state.users[0]}`)}
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
        <CardBox>
          <Card></Card>
        </CardBox>
      </Track>
      <Nav>
        <span class="material-icons">navigate_before</span>
      </Nav>
      <Nav>
        <span class="material-icons">navigate_next</span>
      </Nav>
    </Container>
  )}
}

export default Carousel;