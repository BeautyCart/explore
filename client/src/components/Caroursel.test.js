import React from 'react';
import { shallow, mount } from 'enzyme';


import Carousel from './Carousel';

describe('Carousel', () => {
  it('should render correct in debug mode', () => {
    const component = shallow(<Carousel debug />);
    expect(component).toMatchSnapshot();
  });

});

