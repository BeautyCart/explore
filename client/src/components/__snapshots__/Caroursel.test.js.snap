// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correct in debug mode 1`] = `ShallowWrapper {}`;
